skfda/preprocessing/dim_reduction/feature_extraction/_per_class_feature_transformer.py

```diff
@@ -0,0 +1,205 @@
+    def _validate_transformer(
+        self,
+    ) -> None:
+        """
+        Check that the transformer passed is\
+        scikit-learn-like and that uses target data in fit.
+
+        Args:
+            None
+
+        Returns:
+            None
+        """
+        if not (hasattr(self.transformer, "fit")
+                and hasattr(self.transformer, "transform")
+                or hasattr(self.transformer, "fit_transform")
+                ):
+            raise TypeError(
+                "Transformer should implement fit and "
+                "transform. " + str(self.transformer)
+                + " (type " + str(type(self.transformer)) + ")"
+                " doesn't",
+            )
+
+        tags = self.transformer._get_tags()
+
+        if not (tags['stateless'] and tags['requires_y']):
+            raise TypeError(
+                "Transformer should use target data in fit."
+                + str(self.transformer)
+                + " (type " + str(type(self.transformer)) + ")"
+                " doesn't",
+            )
+
+    def fit(
+        self,
+        X: T,
+        y: ndarray,
+    ) -> PerClassFeatureTransformer:
+        """
+        Fit the model on each class using X as\
+        training data and y as target values.
+
+        Args:
+            X: FDataGrid with the training data.
+            y: Target values of shape = (n_samples).
+
+        Returns:
+            self
+        """
+        classes, class_feature_transformers = _fit_feature_transformer(
+            X,
+            y,
+            self.transformer,
+        )
+
+        self._classes = classes
+        self._class_feature_transformers_ = class_feature_transformers
+
+        return self
+
+    def transform(self, X: T) -> Union[DataFrame, ndarray]:
+        """
+        Transform the provided data using the already fitted transformer.
+
+        Args:
+            X: FDataGrid with the test samples.
+
+        Returns:
+            Eiter array of shape (n_samples, G) or a Data Frame \
+            including the transformed data.
+        """
+        sklearn_check_is_fitted(self)
+        transformed_data = [
+            feature_transformer.transform(X)
+            for feature_transformer in self._class_feature_transformers_
+        ]
+
+        if self.np_array_output:
+            for i in transformed_data:
+                if isinstance(i, FDataGrid or FDataBasis):
+                    raise TypeError(
+                        "There are transformed instances of FDataGrid or "
+                        "FDataBasis that can't be concatenated on a NumPy "
+                        "array.",
+                    )
+            return transformed_data
+
+        if not isinstance(transformed_data[0], FDataGrid or FDataBasis):
+            raise TypeError(
+                "Transformed instance is not of type FDataGrid or"
+                " FDataBasis. It is " + type(transformed_data[0]),
+            )
+
+        frames = [DataFrame(
```

Comment:
I am pretty sure that there is a way to create a `DataFrame` from the columns, without a loop.

Reply:
You are right